phoenix-core/src/main/java/org/apache/phoenix/iterate/RegionScannerFactory.java

```diff
@@ -205,6 +210,22 @@ public boolean nextRaw(List<Cell> result) throws IOException {
              */
             IndexUtil.wrapResultUsingOffset(env, result, offset, dataColumns,
                 tupleProjector, dataRegion, indexMaintainer, viewConstants, ptr);
+
+            byte[] expBytes = scan.getAttribute(BaseScannerRegionObserver.LOCAL_INDEX_FILTER);
+            if (expBytes != null) {
+                ByteArrayInputStream stream = new ByteArrayInputStream(expBytes);
+                DataInputStream input = new DataInputStream(stream);
+                Expression extraWhere = ExpressionType.values()[WritableUtils.readVInt(input)].newInstance();
```

Comment:
This new instance need to be created only once. Here, it is created for each next. We can create the new instance once at the constructor.

Reply:
This is lightweight. But I do agree.

Reply:
There's no constructor since this is an anonymous class, but I can use an instance initializer.

Reply:
I pushed an update using an instance initializer constructing the expression only once.